modules/common-user-vpc/security_groups.tf

```diff
@@ -1,37 +1,35 @@
-resource "aws_security_group" "intra_vpc_and_egress" {
-  description = "allow instances to talk to each other, and have unfettered egress"
+resource "aws_security_group" "intra_vpc_ingress_and_egress" {
+  description = "allow instances to talk to each other, and set up trusted ingress and egress"
   vpc_id      = var.vpc_id
+  name        = "${var.prefix}-intra-cluster-and-trusted-blocks"
 
-  # NOTE: you cannot (should not) mix in-line ingress/egress rules with the
-  # aws_security_group_rule resource
-  # https://www.terraform.io/docs/providers/aws/r/security_group_rule.html
-
-  tags = {
-    Name = "${var.prefix}"
+  ingress {
+    protocol  = "-1"
+    from_port = 0
+    to_port   = 0
+    self      = true
   }
-}
-
-resource "aws_security_group_rule" "intra_vpc_and_egress_ingress_rule" {
-  security_group_id = "${aws_security_group.intra_vpc_and_egress.id}"
 
-  type = "ingress"
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = var.egress_allow_list
+  }
 
-  protocol  = "-1"
-  from_port = 0
-  to_port   = 0
-  self      = true
-}
+  dynamic "ingress" {
```

Comment:
one thing I found while working with this was how confusing it was with the separate resources, and since we're on 0.12, I figured let's wrap every rule into a single SG resource and just let the dynamic block handle the thing it was made for.

Reply:
Good call.